### ISSUE TITLE
Subsequent dynamic import of the same external module result in erroneous substitution of resolution namespace

### ISSUE THREAD

**Issue by usergenic** [rollup/rollup#2487]
- Rollup Version: 0.66.2
- Operating System (or Browser): n/a
- Node Version: 9.11.2 and 10.9.0

### How Do We Reproduce?

Consider the following files:

```js
// app.js
export function loadPage() {
  if (someCondition()) {
    import('./some-module.js');
  }
  if (someOtherCondition) {
    import('./some-module.js');
 }
}
```
```js
// some-module.js
console.log('I have been loaded.');
```

I don't want rollup to inline any of the dynamically imported components, so I provide the fully resolved path to them in my as external option:
```bash
$ rollup app.js --output.format esm --external /resolved/path/to/some-module.js
```

### Expected Behavior

```js
// app.js
function loadPage() {
  if (someCondition()) {
    import('./some-module.js');
  }
  if (someOtherCondition) {
    import('./some-module.js');
 }
}

var app = /*#__PURE__*/Object.freeze({
  loadPage: loadPage
});

export { loadPage };
```

### Actual Behavior

The first instance of the dynamic import of `some-module.js` leaks the local path name in the rolled up result.  I think this called out by https://github.com/rollup/rollup/issues/2481

However, the second instance of the dynamic import of `some-module.js` is replaced with a promise returning the resolution namespace of `app` which is totally bogus.

```js
// app.js
function loadPage() {
  if (someCondition()) {
    import("/Users/brendanb/src/github.com/usergenic/rollup-dynamic-import-problem/some-module.js");
  }
  if (someOtherCondition) {
    Promise.resolve().then(function () { return app; });
 }
}

var app = /*#__PURE__*/Object.freeze({
  loadPage: loadPage
});

export { loadPage };
```

💔

**Comment by usergenic** [comment]
Sidebar: Is there any way to simply disable everything to do with rollup's treatment of dynamic imports and just leave them as-is?  Have been poking at this from lots of angles but am failing to get it to do that.

**Comment by lukastaegert** [comment]
> However, the second instance of the dynamic import of some-module.js is replaced with a promise returning the resolution namespace of app

Are you using experimentalCodeSplitting? Inlining of dynamic imports is the default if this option is not used, which is what you observed here. The rest needs further investigation.

Switching off the handling of dynamic imports is not really possible at the moment but there is a poorly documented plugin hook that may help you: `resolveDynamicImport`

**Comment by usergenic** [comment]
For now I am able to avoid this particular problem using Rollup 0.64.1

I played with resolveDynamicImport but I didn't seem to be able to avoid this behavior when using it.

**Comment by lukastaegert** [comment]
Fix for the first issue at #2505. As for the second issue, see my comment above.
